src/sinks/splunk_hec/logs/integration_tests.rs

```diff
@@ -127,11 +129,9 @@ async fn splunk_insert_message() {
 
     let message = random_string(100);
     let (batch, mut receiver) = BatchNotifier::new_with_receiver();
-    let event = LogEvent::from(message.clone())
-        .with_batch_notifier(&batch)
-        .into();
+    let event = Event::from(message.clone()).with_batch_notifier(&batch);
```

Comment:
I guess I'm not sure what you mean. Always using `.into()`, or doing it implicitly via generic parameters on functions such as `T: Into<Event>`? Instead of using `Event::from(...)`?

Reply:
What I mean is that an `Event` is a generic data type. If you want a log, create a `LogEvent`, etc. From what I understand, `Event::from(String)` is a relic from when there weren't other kinds of events.

Reply:
Ah, yeah, totally.

The reason I did it like this is that `run_and_assert_sink_compliance` takes a `Stream<Item = I> where I: Into<EventArray>`, and there is no conversion to go directly from `LogEvent` to `EventArray` but there was one for `Event`. I can go ahead and add one for `LogEvent` and `Metric`, though, now that I'm thinking about it.

Reply:
Ah, hmm, and doing `LogEvent::from(…).into()` doesn't work because of the double `Into` at work there. The alternate is `Event::Log(LogEvent::from(X))` or `Event::from(LogEvent)` but that could get wordy.

Reply:
Yeah. :/

I ended up addressing this in 6b8a34b9f by simply adding `From` conversions for `LogEvent` (and siblings) into `EventArray`, via `Event::from`.